src/vm/linux/types.rs

```diff
@@ -1 +1,75 @@
 // SPDX-License-Identifier: Apache-2.0
+
+pub const NR_CPUID_CONFIGS: usize = 12;
+
+/// CPUID_CONFIG is designed to enumerate how the host VMM may configure the
+/// virtualization done by the Intel TDX module for a single CPUID leaf and
+/// sub-leaf.
+#[derive(Debug, Default, Clone, Copy)]
+#[repr(C)]
+pub struct CpuidConfig {
+    /// EAX input value to CPUID
+    pub leaf: u32,
+    /// ECX input value to CPUID. A value of -1 indicates a CPUID leaf with
+    /// no sub-leaves.
+    pub sub_leaf: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EAX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub eax: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EBX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub ebx: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in ECX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub ecx: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EDX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub edx: u32,
+}
+
+/// Provides information about the Intel TDX module
```

Comment:
This is equivalent to `struct kvm_tdx_capabilities` in the kernel. We should make note of that.